server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -168,15 +168,18 @@ runHasuraGQ
 runHasuraGQ reqId (query, queryParsed) userInfo resolvedOp = do
   (E.ExecutionCtx logger _ pgExecCtx _ _ _ _ _) <- ask
   (telemTimeIO, respE) <- withElapsedTime $ runExceptT $ case resolvedOp of
-    E.ExOpQuery tx genSql asts -> trace "pg" $ do
+    E.ExOpQuery tx genSql asts -> do
       -- log the generated SQL and the graphql query
       logQueryLog logger query genSql reqId
-      Tracing.interpTraceT id $ executeQuery queryParsed asts genSql pgExecCtx Q.ReadOnly tx
+      trace "Postgres" $
+        Tracing.interpTraceT id $ executeQuery queryParsed asts genSql pgExecCtx Q.ReadOnly tx
 
-    E.ExOpMutation respHeaders tx -> trace "pg" $ do
+    E.ExOpMutation respHeaders tx -> trace "Fetch data" $ do
       logQueryLog logger query Nothing reqId
       ctx <- Tracing.currentContext
-      (respHeaders,) <$> Tracing.interpTraceT (runLazyTx pgExecCtx Q.ReadWrite . withTraceContext ctx . withUserInfo userInfo) tx
+      (respHeaders,) <$>
+        trace "Postgres" do
```

Comment:
This is going to result in two identical-in-duration trace spans, one nested under the other, isn't it? The point is to separate the IO of going to Postgres from the IO of performing remote joins, for example.